sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/GarbageCollector.scala

```diff
@@ -0,0 +1,14 @@
+package scala.scalanative
+package sbtplugin
+
+import java.io.File
+
+/**
+ * @param dir directory name of the gc
+ * @param links dependencies of the gc
+ */
+sealed abstract class GarbageCollector(val dir: String,
+                                       val links: Seq[String] = Nil)
+
+object NoGC extends GarbageCollector("nogc")
+object BoehmGC extends GarbageCollector("boehm", Seq("gc"))
```

Comment:
For style consistency with other ADTs like that (e.g. in NIR), lets structure the code the following way:

```
sealed abstract class GarbageCollector(...)
object GarbageCollector {
  object None extends ...
  object Boehm extends ...
}
```